New package: inspircd-2.0.20

crater2150 opened the pull request:
Accidentally commited to the wrong branch
https://github.com/voidlinux/void-packages/pull/3254

xtraeme commented on srcpkgs/inspircd/template:
```diff
@@ -0,0 +1,53 @@
+configure_args+=" --enable-epoll"
+
+make_dirs="/var/log/inspircd 0755 inspircd inspircd"
+make_dirs+=" /var/run/inspircd 0755 inspircd inspircd"
```
- The `/run` dir should be created by the runit service instead.

xtraeme commented on srcpkgs/inspircd/template:
```diff
@@ -0,0 +1,53 @@
+		make_install_args+=" MODPATH=\"${DESTDIR}/${_libpath}/modules/\""
+		make_install_args+=" CONPATH=\"${DESTDIR}//etc/${pkgname}\""
+
+	${make_cmd} DESTDIR=${DESTDIR} ${make_install_args} ${make_install_target}
```
Those vars are really unnecessary, just set the cmds that this actually needs, i.e `make DESTDIR=...`

xtraeme commented on srcpkgs/inspircd/template:
```diff
@@ -0,0 +1,53 @@
+revision=1
+build_style=configure
+makedepends="gnutls-devel pcre-devel"
+hostmakedepends="gcc-c++ perl pcre-devel pkg-config"
```
- `gcc-c++` is unnecessary, it's always part of the bootstrap pkgs.

xtraeme commented on srcpkgs/inspircd/files/inspircd/run:
```diff
@@ -0,0 +1,7 @@
+#!/bin/sh
+if [ ! -d /var/run/inspircd ]; then
+	mkdir -m0755 -p /var/inspircd
```
`/var/inspircd` vs `/var/run/inspircd` ?

xtraeme commented on srcpkgs/inspircd/files/inspircd/run:
```diff
@@ -0,0 +1,7 @@
+if [ ! -d /var/run/inspircd ]; then
+	mkdir -m0755 -p /var/inspircd
+fi
+exec chpst -u inspircd /usr/bin/inspircd \
```
Please do not hardcode the path to `inspircd`.

Vaelatern commented on srcpkgs/inspircd/template:
```diff
@@ -0,0 +1,46 @@
+revision=1
+build_style=configure
+makedepends="gnutls-devel pcre-devel"
+hostmakedepends="perl pcre-devel pkg-config"
```
`pcre-devel` should go in `makedepends`.

Vaelatern replied:
I believe this is why your cross compliation is currently failing.

crater2150 replied:
It already was in makedepends, but the additional entry in hostmakedepends is not needed, I removed it. But it did not change the cross-compile issue, from what I found googling the error message, it could be some kind of toolchain configuration issue with arm hardfloat: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=62099

Duncaen replied:
`Compiler program:		g++` seems like it doesn't use the right compiler,  it should use `arm-linux-gnueabihf-g++`.

Vaelatern replied:
How would you change this discovery? I've got a package of Inspircd for OpenBSD but OpenBSD doesn't cross compile.

Vaelatern commented on srcpkgs/inspircd/template:
```diff
@@ -0,0 +1,46 @@
+configure_args+=" --binary-dir=${_binpath}"
+configure_args+=" --uid inspircd"
+configure_args+=" --disable-interactive"
+configure_args+=" --enable-gnutls"
```
Just letting you know, inspircd is fully compatible with libressl, so you can, and probably should, enable openssl.

xtraeme replied:
Right, we prefer LibreSSL.

crater2150 commented:
Changed to libressl.
I also switched from the optional pcre-regex-module to the posix-regex-module, as the cross compile fails in m_pcre.so, hope this works around it. The posix one also does not have additional dependencies.
